src/handlers.rs

```diff
@@ -33,35 +33,34 @@ use super::protocol::RequestType;
 ///
 /// * `log` - A `slog` Logger.
 /// * `request` - The request to handle.
-/// * `send_slice` - A callback that will be used to send bytes back to the client, if
-///   there is a response to send.
-pub fn handle_request<F>(log: &Logger, request: &protocol::Request, send_slice: F) -> Result<()>
-where
-    F: FnMut(&[u8]) -> Result<()>,
-{
+pub fn handle_request(log: &Logger, request: &protocol::Request) -> Result<Vec<u8>> {
     debug!(log, "handling request"; "request" => ?request);
     match request.ty {
         RequestType::GETPWBYUID => {
             let key = CStr::from_bytes_with_nul(request.key)?;
             let uid = atoi::<u32>(key.to_bytes()).context("invalid uid string")?;
             let user = User::from_uid(Uid::from_raw(uid))?;
-            send_user(log, user, send_slice)
+            debug!(log, "got user"; "user" => ?user);
```

Comment:
for debug, might be useful to log which libc method we're using for the lookup

```rust
debug!(log, "User::from_uid"; "user" => ?user)
```

this is what @geofft wanted with `stringify!`. does slog have an equivalent of [`dbg`](https://doc.rust-lang.org/std/macro.dbg.html)?

Reply:
I don't think it has a `dbg!` equivalent, but I did attach the method as context up in `handle_stream`